File: site/php/ItemBuyOrder.php
<?php

$dir = "5.11-EUW-RANKED/*";




$itemTotalMatches = array();
$itemLaneCount = array();


foreach(glob($dir) as $file)  { 
$string = file_get_contents($file);
     $json = json_decode($string, true);

     for($i = 0; $i < 10; $i++) {
           $lane = $json['participants'][$i]['timeline']['lane'];
             $championLane = array(
        "0" => 0,
        "1" => 0,
        "2" => 0,
        "3" => 0,
        "4" => 0,
        "5" => 0
    );
         
          for($j = 0; $j < 6; $j++) {
         
        $itemId = $json['participants'][$i]['stats']['item' . strval($j)];
      
         
            
        if (!array_key_exists($itemId, $itemTotalMatches)) {
                $itemTotalMatches[$itemId] = 1;
                $itemLaneCount[$itemId] = $championLane;
                
         } else { 
             $itemTotalMatches[$itemId] += 1;
         }
         
         $itemLaneCount[$itemId][$j] += 1;
     }
         
     }
}

$championLanePercent = $itemLaneCount;


foreach ($itemLaneCount as $championId => $laneArray) {
    foreach ($laneArray as $lane => $laneCount) {
        $championLanePercent[$championId][$lane] = round(($laneCount/$itemTotalMatches[$championId])*100,0); 
    }
}
    

print_r($championLanePercent);


?>

File: scripts/ItemBuyOrder.py
import os, os.path, json,operator


matches = 0;


championTotalMatches = {}
championLaneCount = {}

for root, _, files in os.walk("5.11-EUW-RANKED"):
    for f in files:
        matches += 1
        print(matches)
        fullpath = os.path.join(root, f)
        if ".json" in fullpath:
            with open(fullpath) as data_file:

                jsonData = json.load(data_file)
                for x in range(0, 10):
                         championLane = {"0": 0,
                                    "1":0,
                                    "2" : 0,
                                    "3" : 0,
                                    "4" : 0,
                                     "5" : 0}

                         for y in range(0, 6):


                             championId = jsonData["participants"][x]["stats"]["item" + str(y)]


                             if championId not in championTotalMatches:
                                 championTotalMatches[championId] = 1
                                 championLaneCount[championId] = championLane
                             else:
                                 championTotalMatches[championId] += 1


                             championLaneCount[championId][str(y)] += 1


championLanePercent = {}


for championId,laneArray  in championLaneCount.items():

        championLanePercent[championId] = max(laneArray, key=lambda x: laneArray[x[0]])

array = {}

def find_between( s, first, last ):
    try:
        start = s.index( first ) + len( first )
        end = s.index( last, start )
        return s[start:end]
    except ValueError:
        return ""

with open("items.json") as data_file:

        jsonData = json.load(data_file)

for itemId,useless in championLaneCount.items():
    try:
       description = jsonData["data"][str(itemId)]["description"];
       stats = find_between(description,"stats>","</stats")
       split = stats.split("<br>")
       for x in range(0,len(split)):
           statWord = split[x].replace("<mana>","",1).replace("</mana>","",1)
           print(statWord + " | " + jsonData["data"][str(itemId)]["name"])



    except Exception:
        pass



for v in array:
    print (str(v) + ",")




#for key, value in championLanePercent.items():
   # print(("\"" + str(key) + "\" => " + str(value) + ",").replace(":"," => ").replace("{","array (").replace("}",")"))
